Bottom: b2e098e188186792d7c82fa2a309e47c26d4ccae
Top:    28253925f4492188054632e5a7067a104adb4106
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-01 17:10:53 +0000

Play with grid snapping heuristics a little

Require the pointer to be inside the bounding box of a pin / pad / via before snapping.
This makse it MUCH easier to place vias in a regular grid spaced at the PCB grid spacing.


---

diff --git a/src/crosshair.c b/src/crosshair.c
index 6ae12e9..290d632 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -993,7 +993,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PAD_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1050,7 +1050,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (PCB->RatDraw || TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (PIN_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid self-snapping when moving */
   if (ans != NO_TYPE &&
@@ -1068,7 +1068,7 @@ FitCrosshairIntoGrid (Coord X, Coord Y)
   ans = NO_TYPE;
   if (TEST_FLAG (SNAPPINFLAG, PCB))
     ans = SearchObjectByLocation (VIA_TYPE, &ptr1, &ptr2, &ptr3,
-                                  Crosshair.X, Crosshair.Y, PCB->Grid / 2);
+                                  Crosshair.X, Crosshair.Y, 0);
 
   /* Avoid snapping vias to any other vias */
   if (Settings.Mode == MOVE_MODE &&
